src/components/dev-hub/star-rating.js

```diff
@@ -0,0 +1,213 @@
+import React, { useCallback, useReducer } from 'react';
+import PropTypes from 'prop-types';
+import styled from '@emotion/styled';
+import StarIcon from '~components/dev-hub/icons/star-icon';
+import Button from '~components/dev-hub/button';
+
+import { size, fontSize, screenSize, lineHeight } from './theme';
+
+const StyledButton = styled(Button)`
+    margin-right: ${size.mediumLarge};
+    padding: 0;
+
+    display: inline-flex;
+    justify-content: center;
+    align-items: center;
+
+    svg {
+        height: ${size.mediumLarge};
+        width: ${size.mediumLarge};
+    }
+
+    @media ${screenSize.mediumAndUp} {
+        margin: ${({ isActive }) => (isActive ? '0' : '0 4px')};
+        padding: 4px;
+        svg {
+            height: auto;
+            width: auto;
+        }
+    }
+`;
```

Comment:
One nitpick I have is to keep the CSS alphabetized, could we please reorder some of these properties? (Leaving media queries at the end is good)? Thanks!

Reply:
Thanks for help! I will fix it and merge PR after checking.